WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -922,6 +946,18 @@ open class ReaderDetailViewController: UIViewController, UIViewControllerRestora
 
     // MARK: - Actions
 
+    @IBAction func didTapSaveForLaterButton(_ sender: UIButton) {
+        guard let readerPost = post, let context = readerPost.managedObjectContext else {
+            return
+        }
+
+        SaveForLaterAction().execute(with: readerPost, context: context) {
+
+        }
+
+        saveForLaterButton.isSelected = readerPost.isSavedForLater
```

Comment:
Should we move this into the completion block for the save for later action? Otherwise the flag might not have been toggled by the time we execute this code?